lib/card/set_pattern.rb

```diff
@@ -130,6 +135,34 @@ def rule_set_key
         [ @anchor_id, self.class.pattern_code ].map( &:to_s ) * '+'
       end
     end
-  
+  end
+
+  class TypeSet < SetPattern
+    def initialize card
+      super
+      @inherit_card = card unless module_key
+    end
+
+    def lookup_inherited_key
+      return unless @inherit_card
+      card, @inherit_card = @inherit_card, nil
```

Comment:
Seemed like we wanted to keep the card out of these objects, but I was getting infinite recursion through rule_card if I did it when initializing the SetPattern instance.  We'll use it the first time this method is called where it will also clear the attribute this forgetting the card.  I guess the instance is really for that specific card.  The remnant of that will be in the return value to be stored in @inherited_key.  This could be any rule with the current code (two lines commented out would restrict it to type rules).